addons/binding/org.openhab.binding.spotify/src/main/java/org/openhab/binding/spotify/internal/SpotifyAuthServlet.java

```diff
@@ -0,0 +1,202 @@
+     *
+     * @param playerTemplate The player template to format the player values in
+     * @param servletBaseURL the redirect_uri to be used in the authorization url created on the authorization button.
+     * @return A String with the players formatted with the player template
+     */
+    private String formatPlayers(String playerTemplate, String servletBaseURL) {
+        final List<SpotifyAccountHandler> players = spotifyAuthService.getSpotifyAccountHandlers();
+
+        return players.isEmpty() ? HTML_EMPTY_PLAYERS
+                : players.stream().map(p -> formatPlayer(playerTemplate, p, servletBaseURL))
+                        .collect(Collectors.joining());
+    }
+
+    /**
+     * Formats the HTML of a Spotify Bridge Player and returns it as a String
+     *
+     * @param playerTemplate The player template to format the player values in
+     * @param handler The handler for the player to format
+     * @param servletBaseURL the redirect_uri to be used in the authorization url created on the authorization button.
+     * @return A String with the player formatted with the player template
+     */
+    private String formatPlayer(String playerTemplate, SpotifyAccountHandler handler, String servletBaseURL) {
+        final Map<String, String> map = new HashMap<>();
+
+        map.put(PLAYER_ID, handler.getUID().getAsString());
+        map.put(PLAYER_NAME, handler.getLabel());
+        final String spotifyUser = handler.getUser();
+
+        if (handler.isAuthorized()) {
+            map.put(PLAYER_AUTHORIZED_CLASS, " authorized");
+            map.put(PLAYER_SPOTIFY_USER_ID, String.format(" (Authorized user: %s)", spotifyUser));
+        } else if (!StringUtil.isBlank(spotifyUser)) {
+            map.put(PLAYER_AUTHORIZED_CLASS, " Unauthorized");
+            map.put(PLAYER_SPOTIFY_USER_ID, String.format(" (Unauthorized user: %s)", spotifyUser));
+        } else {
+            map.put(PLAYER_AUTHORIZED_CLASS, "");
+            map.put(PLAYER_SPOTIFY_USER_ID, "");
+        }
+
+        map.put(PLAYER_AUTHORIZE, handler.formatAuthorizationUrl(servletBaseURL));
+        return replaceKeysFromMap(playerTemplate, map);
+    }
+
+    /**
+     * Replaces all keys from the map found in the template with values from the map. If the key is not found the key
+     * will be kept in the template.
+     *
+     * @param template template to replace keys with values
+     * @param map map with key value pairs to replace in the template
+     * @return a template with keys replaced
+     */
+    private String replaceKeysFromMap(String template, Map<String, String> map) {
+        final Matcher m = MESSAGE_KEY_PATTERN.matcher(template);
+        final StringBuffer sb = new StringBuffer();
+
+        while (m.find()) {
+            try {
+                final String key = m.group(1);
+                m.appendReplacement(sb, Matcher.quoteReplacement(map.getOrDefault(key, "${" + key + '}')));
+            } catch (RuntimeException e) {
```

Comment:
Do we have to catch this broad especially here I would expect that the range of exceptions is small

Reply:
Did you address this one?

Reply:
Same comment as above. It's in a servlet. This just catches all unchecked exceptions, like NullPointers. Normally those are not catched as it's unknown what is thrown unlike checked exceptions. So I would have to guess what exceptions would be thrown. This is just very defensive programming to avoid showing mangled pages to the user whatever happens.